src/source/source_cache.js

```diff
@@ -291,7 +294,10 @@ class SourceCache extends Evented {
         const heightInTiles = Math.ceil(transform.height / transform.tileSize) + 1;
         const approxTilesInView = widthInTiles * heightInTiles;
         const commonZoomRange = 5;
-        this._cache.setMaxSize(Math.floor(approxTilesInView * commonZoomRange));
+
+        const maxSize = this._tileCacheSize || Math.floor(approxTilesInView * commonZoomRange);
```

Comment:
Could we change this logic to:
```js
const viewDependentMaxSize = Math.floor(approxTilesInView * commonZoomRange);
const maxSize = typeof this._tileCacheSize === 'number' ? Math.min(this._tileCacheSize, viewDependentMaxSize) : viewDependentMaxSize;
```

This way, the new option would be setting an upper bound on the cache size, rather than forcing it to be fixed.